packages/esm-patient-common-lib/src/workspaces/workspaces.ts

```diff
@@ -325,3 +319,73 @@ function getUpdatedWorkspaceWindowState(workspaceAtTop: OpenWorkspace) {
 export function resetWorkspaceStore() {
   getWorkspaceStore().setState(initialState);
 }
+
+export function closeAllWorkspaces(onClosingWorkspaces: () => void = () => {}) {
+  const store = getWorkspaceStore();
+
+  const canCloseAllWorkspaces = store.getState().openWorkspaces.every(({ name }) => {
+    const canCloseWorkspace = getWhetherWorkspaceCanBeClosed(name);
+    return canCloseWorkspace;
+  });
+
+  const updateWorkspaceStore = () => {
+    resetWorkspaceStore();
+    onClosingWorkspaces?.();
+  };
+
+  if (!canCloseAllWorkspaces) {
+    showWorkspacePrompts('closing-all-workspaces', updateWorkspaceStore);
+  } else {
+    updateWorkspaceStore();
+  }
+}
+
+export function getWhetherWorkspaceCanBeClosed(name: string, ignoreChanges = false) {
+  const promptCheckFcn = getPromptBeforeClosingFcn(name);
+  return ignoreChanges || !promptCheckFcn || !promptCheckFcn();
+}
```

Comment:
@vasharma05 do you mind, explaining how this works?
While doing some tests, I didn't find where it returned a false i.e for workspace can't close
Thanks.

Reply:
To get the workspace to be closed, you will have to add the prompt function in the individual workspaces, which I am looking in the O3-1760.

For testing purpose, directly return `false` from the function